profiles/fscloud/variables.tf

```diff
@@ -0,0 +1,153 @@
+
+variable "resource_group_id" {
+  description = "ID of resource group to use when creating the mongodb database"
+  type        = string
+}
+
+variable "mongodb_version" {
+  description = "The version of mongodb. If null, the current default ICD mongodb version is used."
+  type        = string
+  default     = null
+}
+
+variable "plan" {
+  type        = string
+  description = "The name of the service plan that you choose for your mongodb instance"
+  default     = "standard"
+}
+
+variable "region" {
+  description = "The region mongodb is to be created on. The region must support BYOK ( us-south, us-east, and eu-de)"
+  type        = string
+  default     = "us-south"
+  validation {
+    condition = anytrue([
+      var.region == "us-south",
+      var.region == "us-east",
+      var.region == "eu-de"
+    ])
+    error_message = "region must be in a BYOK location, us-south, us-east or eu-de"
+  }
+}
+
+variable "allowlist" {
+  description = "(Optional, List of Objects) A list of allowed IP addresses for the database."
+  type = list(object({
+    address     = string
+    description = string
+  }))
+  default = []
+}
+
+variable "configuration" {
+  description = "Database Configuration in JSON format."
+  type = object({
+    maxmemory                   = optional(number)
+    maxmemory-policy            = optional(string)
+    appendonly                  = optional(string)
+    maxmemory-samples           = optional(number)
+    stop-writes-on-bgsave-error = optional(string)
+  })
+  default = null
+}
+
+variable "cpu_count" {
+  description = "Number of CPU cores available to the mongodb instance"
+  type        = number
+  default     = 7
+}
+
+variable "memory_mb" {
+  description = "Memory available to the mongodb instance"
+  type        = number
+  default     = 1024
+}
+
+variable "disk_mb" {
+  description = "Disk space available to the mongodb instance"
+  type        = number
+  default     = 20480
+}
+
+
+variable "instance_name" {
+  description = "Name of the new mongodb instance"
+  type        = string
+}
+
+variable "tags" {
+  type        = list(any)
+  description = "Optional, Array of Strings) A list of tags that you want to add to your instance."
+  default     = []
+}
+
+variable "key_protect_key_crn" {
+  type        = string
+  description = "The root key CRN of a Key Management Service like Key Protect or Hyper Protect Crypto Service (HPCS) that you want to use for disk encryption. See https://cloud.ibm.com/docs/cloud-databases?topic=cloud-databases-key-protect&interface=ui#key-byok for current list of supported regions for BYOK"
+}
+
+variable "backup_encryption_key_crn" {
+  type        = string
+  description = "The CRN of a key protect key, that you want to use for encrypting disk that holds deployment backups. If null, will use 'key_protect_key_crn' as encryption key. If 'key_protect_key_crn' is also null database is encrypted by using randomly generated keys."
+  default     = null
```

Comment:
I don't think null should be allowed in fscloud as the profile should force users to BYOK right @daniel-butler-irl ?

Reply:
need to remove
> If 'key_protect_key_crn' is also null database is encrypted by using randomly generated keys.
as the encryption crn is a requirement, so we always use one of the provided crns